File: spring_java_orm_sample/src/main/java/com/example/demo/mybatis/typehandler/BookStockMovementTypeHandler.java
package com.example.demo.mybatis.typehandler;

import com.example.demo.data.domain.BookStockMovementType;
import org.apache.ibatis.type.BaseTypeHandler;
import org.apache.ibatis.type.JdbcType;
import org.apache.ibatis.type.MappedTypes;

import java.sql.CallableStatement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

@MappedTypes(BookStockMovementType.class)
public class BookStockMovementTypeHandler extends BaseTypeHandler<BookStockMovementType> {
    @Override
    public void setNonNullParameter(PreparedStatement ps, int i, BookStockMovementType parameter, JdbcType jdbcType) throws SQLException {
        ps.setInt(i, parameter.getValue());
    }

    @Override
    public BookStockMovementType getNullableResult(ResultSet rs, String columnName) throws SQLException {
        final var value = rs.getInt(columnName);
        return rs.wasNull() ? null : BookStockMovementType.of(value);
    }

    @Override
    public BookStockMovementType getNullableResult(ResultSet rs, int columnIndex) throws SQLException {
        final var value = rs.getInt(columnIndex);
        return rs.wasNull() ? null : BookStockMovementType.of(value);
    }

    @Override
    public BookStockMovementType getNullableResult(CallableStatement cs, int columnIndex) throws SQLException {
        final var value = cs.getInt(columnIndex);
        return cs.wasNull() ? null : BookStockMovementType.of(value);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/mybatis/typehandler/BookStockMovementTypeHandler.kt
package com.example.demo.mybatis.typehandler

import com.example.demo.data.domain.BookStockMovementType
import org.apache.ibatis.type.BaseTypeHandler
import org.apache.ibatis.type.JdbcType
import org.apache.ibatis.type.MappedTypes
import java.sql.CallableStatement
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException

@MappedTypes(BookStockMovementType::class)
class BookStockMovementTypeHandler : BaseTypeHandler<BookStockMovementType?>() {
    @Throws(SQLException::class)
    override fun setNonNullParameter(
        ps: PreparedStatement,
        i: Int,
        parameter: BookStockMovementType?,
        jdbcType: JdbcType?
    ) {
        ps.setInt(i, parameter!!.value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnName: String?): BookStockMovementType? {
        val value = rs.getInt(columnName)
        return if (rs.wasNull()) null else BookStockMovementType.of(value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(rs: ResultSet, columnIndex: Int): BookStockMovementType? {
        val value = rs.getInt(columnIndex)
        return if (rs.wasNull()) null else BookStockMovementType.of(value)
    }

    @Throws(SQLException::class)
    override fun getNullableResult(cs: CallableStatement, columnIndex: Int): BookStockMovementType? {
        val value = cs.getInt(columnIndex)
        return if (cs.wasNull()) null else BookStockMovementType.of(value)
    }
}
